### ISSUE TITLE
Setting objects' attributes in vakt

### ISSUE THREAD

**Issue by anton-khodak** [kolotaev/vakt#2]
Hi Egor,

I'm looking into your library, and I would like to understand is it possible to attach attributes to entities as per ABAC model (https://en.wikipedia.org/wiki/Attribute-based_access_control#Attributes)? So far, I see that I can use contextual attributes in Roles, but I can't see how I can add subject, object and resources attributes.

 For example, how do I implement the following simple scenario with vakt: I have users that register continuously, a user has an attribute "collaborator" with a list of dataset ids, when a new id is added to this list, they can update a dataset with this id.

**Comment by kolotaev** [comment]
Hi @anton-khodak,

Generally speaking vakt is not as opinionated as other ABAC implementations found in the wild in defining attributes. It's more a "policy-based access control" part of ABAC definition per se. But from another perspective it's somewhat more flexible since it allows you to build your own vision on setting attributes.

Answering your question (2 variants since I haven't clearly understood the scenario), one possible way to do it with vakt:
You can create a Policy like:

```python
    Policy(
        uid=12345,
        effect=ALLOW_ACCESS,
        subjects=['username=<.*>;position=<owner|collaborator>'], # `;` and `=` here are arbitrary delimiters
        resources=('some-resource-definition'),
        actions=['<read|update>'],
        rules={
            'dataset-ids': InList(['a', 'b', 'c', 'd', 'e']),
        },
    )
```
and build an Inquiry:
```python
Inquiry(subject='username=max;position=collaborator', action='read', context={'dataset-ids':'d'})
```
where `AnyInList` is a custom rule that is, for example, satisfied when 'd' can be found in ['a', 'b', 'c', 'd', 'e'].

Every time you add an additional ID to your system you should update the corresponding '12345' Policy 'dataset-ids' rule with a new list of IDs that is [old-list] + new-id.

If your resource is the dataset-IDs you are talking about, then possible way:
```python
    Policy(
        uid=12345,
        effect=ALLOW_ACCESS,
        subjects=['username=<.*>;position=<owner|collaborator>'], # `;` and `=` here are arbitrary delimiters
        resources=['dataset:1', 'dataset:2'], # imagine 2 possible datasets each with some information like IDs
        actions=['dataset:id:<add|remove>', 'delete'] # add/remove id from dataset; delete - action that deletes the whole dataset
    )
```
```python
Inquiry(subject='username=max;position=collaborator', action='dataset:id:add', resource='dataset:1') # allowed

Inquiry(subject='username=jerry;position=contributor', action='dataset:id:add', resource='dataset:1') # rejected

Inquiry(subject='username=kim;position=owner', action='dataset:id:remove', resource='dataset:3') # rejected
```

**Comment by kolotaev** [comment]
Is planned be implemented in https://github.com/kolotaev/vakt/issues/4.

**Comment by kolotaev** [comment]
@anton-khodak, you probably might be interested:
The functionality we've discussed here is implemented in vakt >= 1.2.
See examples and updated documentation.

Thanks for the issue.
